src/types/AuditLogEvent.ts

```diff
@@ -0,0 +1,42 @@
+export default class AuditLogEvent {
+  private constructor(private code: string, private description: string) {}
+
+  public static readonly loggedIn = new AuditLogEvent("user-logged-in", "User logged in")
+
+  public static readonly changedPassword = new AuditLogEvent("user-password.changed", "Change password")
+
+  public static readonly deletedUser = new AuditLogEvent("user-deletion.deleted", "Delete user")
+
+  public static readonly resetPassword = new AuditLogEvent("user-password.reset", "Reset password")
+
+  public static readonly createdUser = new AuditLogEvent("user-creation.created", "Create user")
+
+  public static readonly failedCreatingNewEmailWhileCreatingUser = new AuditLogEvent(
+    "user-creation.failed-email",
+    "Error creating new user email"
+  )
+
+  public static readonly failedNotificationWhileCreatingUser = new AuditLogEvent(
+    "user-creation.failed-notification",
+    "Error sending notification email of new user creation"
+  )
+
+  public static readonly failedToEmailUserWhileCreatingUser = new AuditLogEvent(
+    "user-creation.failed-emailing-user",
+    "Error sending email to new user"
+  )
+
+  public static readonly editedUser = new AuditLogEvent("user-edit.updated", "Edit user")
+
+  getCode() {
```

Comment:
Do we actually need getters for these? They're public attributes

Reply:
The `code` and `description` fields are private at the moment but I can make them `public readonly` in constructor signature to achieve the same thing and remove `getCode` and `getDescription` functions.
I'll update it.